src/Build/BackEnd/Components/RequestBuilder/TargetEntry.cs

```diff
@@ -748,7 +748,20 @@ internal void EnterLegacyCallTargetScope(Lookup lookup)
         /// </summary>
         internal void MarkForError()
         {
-            ErrorUtilities.VerifyThrow(_state != TargetEntryState.Completed, "State must not be Completed. State is {0}.", _state);
+            // In the case that the Target was skipped due to a condition but a dependent target failed subsequent execution
+            // must stop. Detect that case and set the WorkUnitActionCode to stop and mark the TargetEntry for ErrorExecution.
```

Comment:
So:
* We're in this method because a dependent target failed
* But this target is already marked completed/skipped/continue

And therefore we want to update the current status to completed/skipped/stop?